src/ast/mod.rs

```diff
@@ -1047,6 +1054,27 @@ pub enum Statement {
         unique: bool,
         if_not_exists: bool,
     },
+    /// CREATE ROLE
+    CreateRole {
+        names: Vec<ObjectName>,
+        if_not_exists: bool,
+        // Postgres
+        login: Option<bool>,
+        inherit: Option<bool>,
+        bypassrls: Option<bool>,
+        password: Option<Password>,
+        superuser: Option<bool>,
+        create_db: Option<bool>,
+        create_role: Option<bool>,
+        replication: Option<bool>,
+        connection_limit: Option<Expr>,
+        valid_until: Option<Expr>,
```

Comment:
One downside with this implementation is that the order of the grants is not preserved and thus the original SQL can not be reconstructed

What do you think about an alternate encoding like:

```rust
...
role_grants: Vec<RoleGrant>
...
```

And then


```rust
enum RoleGrant {
  Login, 
  Inherit,
  Password(Password)
   ..
}
```

However, I see this PR follows the model of GRANT so I think it is also fine as is